tests/experimental/datasources/conftest.py

```diff
@@ -115,3 +123,139 @@ def inject_engine_lookup_double(monkeypatch: MonkeyPatch) -> ExecutionEngineDoub
     finally:
         for source, engine in original_engine_override.items():
             source.execution_engine_override = engine
+
+
+def pandas_data(
+    context: AbstractDataContext,
+) -> tuple[AbstractDataContext, Datasource, DataAsset, BatchRequest]:
+    csv_path = (
+        pathlib.Path(__file__) / "../../../test_sets/taxi_yellow_tripdata_samples"
+    ).resolve()
+    pandas_ds = context.sources.add_pandas(name="my_pandas")
+    asset = pandas_ds.add_csv_asset(
+        name="csv_asset",
+        base_directory=csv_path,
+        regex=r"yellow_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2})\.csv",
+        order_by=["year", "month"],
+    )
+    batch_request = asset.get_batch_request({"year": "2019", "month": "01"})
+    return context, pandas_ds, asset, batch_request
```

Comment:
Should these be fixtures by the way?

Reply:
@cdkini This function, and those for the other backends and their multi-batch counterparts, are all used in fixtures.  However, making them -- in addition -- fixtures themselves causes problems with `pytest` which is notoriously known for not accepting fixtures as parameters to other fixtures.  (I tried, and after error messages explaining the above, reverted to just declaring them as methods to be used in fixtures.)  If you know of a way, please do let me know.  Thank you.